scheds/rust/scx_layered/src/main.rs

```
@@ -618,6 +625,11 @@ struct Opts {
 
     /// Layer specification. See --help.
     specs: Vec<String>,
+
+    /// Periodically tasks to reevaluate which layer they belong to. Default period of 0
+    /// turns this off.
+    #[clap(long, default_value = "0")]
+    force_layer_refresh_ms: u64,
 }
```

Why not tie this to the runtime matches being used? This is not useful for other matches but is critical for the runtime matches. It's unnecessarily error-prone to make this an independent option.

---

Sounds good, I will make it so that this only triggers on layers with AvgRuntime rules and adjust the name accordingly. I initially had that and removed, my rationale was it was that silently ignoring the flag for some layers would be unexpected.

---

Maybe it can be named in a way that clarifies where it applies. Also, it'd be better if the users aren't required to set this option to use the runtime matches. ie. If you limit its scope to the ones that need it, you can also pick a default value that makes sense for them instead of 0.